src/Bicep.Core/SemanticModel/NameBindingVisitor.cs

```diff
@@ -110,30 +131,61 @@ private Symbol ValidateFunctionFlags(Symbol symbol, TextSpan span)
             return symbol;
         }
 
-        private Symbol LookupSymbolByName(string name, TextSpan span)
+        private Symbol LookupSymbolByName(string name, TextSpan span, string? @namespace)
         {
-            if (this.declarations.TryGetValue(name, out var localSymbol))
+            NamespaceSymbol? FindNamespace(string name) =>
+                this.namespaces
+                .Where(ns => ns.Name.Equals(name, LanguageConstants.IdentifierComparison))
+                .FirstOrDefault();
+
+            Symbol? foundSymbol;
+            if (string.IsNullOrEmpty(@namespace))
```

Comment:
We should just use explicit null checks (e.g. `if (@namespace != null)`)

Reply:
Yup, updated.